app/vminsert/netstorage/netstorage.go

```diff
@@ -319,11 +319,15 @@ var cannotCloseStorageNodeConnLogger = logger.WithThrottler("cannotCloseStorageN
 
 var cannotSendBufsLogger = logger.WithThrottler("cannotSendBufRows", 5*time.Second)
 
-func sendToConn(bc *handshake.BufferedConn, buf []byte) error {
+func sendToConnBuf(bc *handshake.BufferedConn, buf []byte) error {
```

Comment:
maybe we don't need this func at all? 

the bc is checked for being nil right before the call of this function on L:282. It has no other usages.

Reply:
function `sendToConnBuf` checks len of buf. If it's zero, fast path trigger with early return.

function `sendToConn` do not perform this check anymore and may send an empty buf. 

 It's needed for vmstorage read-only check . function `checkReadOnlyMode` uses `sendToConn`  with zero len buf for reading state response from vmstorage.

Reply:
Yes, but if you remove only `!=nil` check from `sendToConn` - it should do the thing, right?

Reply:
Yep, but it may have ingestion performance impact.

Reply:
why?

Reply:
I've double checked vminsert code and the same check already performed by the code  at `(sn *storageNode) run()` method:
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/cluster/app/vminsert/netstorage/netstorage.go#L180

 Simplified fix.